cmd/odf/ceph/purge_osd.go

```diff
@@ -0,0 +1,65 @@
+package rook
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/red-hat-storage/odf-cli/cmd/odf/root"
+	"github.com/red-hat-storage/odf-cli/pkg/rook/osd"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	"github.com/rook/kubectl-rook-ceph/pkg/mons"
+	"github.com/rook/kubectl-rook-ceph/pkg/rook"
+	"github.com/spf13/cobra"
+)
+
+var forceFlag = "force"
+
+var cephPurgeOsdCmd = &cobra.Command{
+	Use:     "purge-osd",
+	Short:   "Permanently remove an OSD from the cluster.",
+	Args:    cobra.ExactArgs(1),
+	Example: "odf ceph purge-osd <id>",
+	PreRunE: validateOsdID,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+		clientsets := root.GetClientsets(ctx)
+		forceflagValue := cmd.Flag(forceFlag).Value.String()
+
+		forceOSDRemovalBool, err := strconv.ParseBool(forceflagValue)
+		if err != nil {
+			logging.Fatal(errors.Wrapf(err, "failed to parse --force-osd-removal flag"))
+		}
+
+		safeToDestroy, err := osd.SafeToDestroy(ctx, clientsets, root.OperatorNamespace, root.StorageClusterNamespace, args[0])
+		if !safeToDestroy && !forceOSDRemovalBool {
+			logging.Fatal(errors.Wrapf(err, "osd.%s is not safe to destroy", args[0]))
+		} else if !safeToDestroy && forceOSDRemovalBool {
+			var answer string
+			logging.Warning("Are you sure you want to purge-osd osd.%s? This will possible lead to data loss. If yes enter 'yes-i-really-want-to-destroy'", args[0])
```

Comment:
```suggestion
			logging.Warning("Are you sure you want to purge osd.%s? The OSD is *not* safe to destroy. This may lead to data loss. If you are sure the OSD should be purged, enter 'yes-force-destroy-osd'", args[0])
```